presto-main/src/main/java/com/facebook/presto/server/QueryResource.java

```diff
@@ -100,11 +101,21 @@ public void getAllQueryInfo(
             proxyResponse(servletRequest, asyncResponse, xForwardedProto, uriInfo);
             return;
         }
+        if (limitFilter == null) {
+            limitFilter = dispatchManager.getMaxQueriesToReturn();
+        }
         QueryState expectedState = stateFilter == null ? null : QueryState.valueOf(stateFilter.toUpperCase(Locale.ENGLISH));
         ImmutableList.Builder<BasicQueryInfo> builder = new ImmutableList.Builder<>();
+        // TODO: With us having 'limit' now, it would be nice to have 'basic query info' sorted  by some time
+        // (creation time?) in the query list that we get from dispatch manager.
+        // At the moment I don't see how we can add DateTime member to 'basic query info' and not break something.
         for (BasicQueryInfo queryInfo : dispatchManager.getQueries()) {
             if (stateFilter == null || queryInfo.getState() == expectedState) {
                 builder.add(queryInfo);
+                --limitFilter;
+                if (limitFilter <= 0) {
```

Comment:
The caveat of checking after adding is that no matter what the `limitFilter` is, we will return one query at least. And if limit is `Integer.MIN_VALUE`, it will underflow and return pretty much everything. Instead, should we check this limitFilter before checking if a query is eligible?

Reply:
>> that no matter what the limitFilter is, we will return one query at least.

Yes, that's the intent. Kind of silly to call this endpoint and not having any queries coming back (status filter notwithstanding).

We need to check the limit after checking is query is eligible, otherwise we can exhaust limit on non-eligible queries and return nothing.

Reply:
Right. Let's check for the input and throw if the limit is not positive, and then we don't need to worry about stuff here.